projects/igniteui-angular/src/lib/query-builder/query-builder.component.spec.ts

```diff
@@ -1572,7 +1572,125 @@ describe('IgxQueryBuilder', () => {
       QueryBuilderFunctions.verifyExpressionChipContent(fix, [1], 'OrderId', 'Greater Than', '3');
     }));
 
-    it('Should not make bug where existing inner query is leaking to a newly created one', fakeAsync(() => {
+    it(`Should focus edited expression chip after click on the 'commit'/'discard' button.`, fakeAsync(() => {
+      queryBuilder.expressionTree = QueryBuilderFunctions.generateExpressionTree();
+      fix.detectChanges();
+      tick(100);
+      fix.detectChanges();
+
+      // Click the 'OrderId' chip to enter edit mode.
+      QueryBuilderFunctions.clickQueryBuilderTreeExpressionChip(fix, [1]);
+      tick(100);
+      fix.detectChanges();
+
+      // Click on the 'commit' button
+      const commitBtn = QueryBuilderFunctions.getQueryBuilderExpressionCommitButton(fix);
+      commitBtn.click();
+      fix.detectChanges();
+      tick(100);
+      fix.detectChanges();
+      // Verify focused chip
+      QueryBuilderFunctions.verifyFocusedChip('OrderId', 'Greater Than', '3');
+
+      // Click the 'OrderId' chip to enter edit mode.
+      QueryBuilderFunctions.clickQueryBuilderTreeExpressionChip(fix, [1]);
+      tick(100);
+      fix.detectChanges();
+
+      // Click on the 'discard' button
+      const closeBtn = QueryBuilderFunctions.getQueryBuilderExpressionCloseButton(fix);
+      closeBtn.click();
+      fix.detectChanges();
+      tick(100);
+      fix.detectChanges();
+      // Verify focused chip
+      QueryBuilderFunctions.verifyFocusedChip('OrderId', 'Greater Than', '3');
+    }));
+
+    it(`Should focus proper expression chip after switching edit mode and click on the 'commit'/'discard' button.`, fakeAsync(() => {
```

Comment:
This test seems a bit odd, do we need it at all?

Reply:
I have added it because there is an issue with this behavior that was fixed